### PR TITLE
[Blocked] Feature/368: Add 202 Response Integration Tests for Async Handlers

### PR DESCRIPTION
**Pull request by ramanmangla**
In reference to tickets #368 .

 - This adds integration tests for 202 response from the merged handlers to verify correct working.
 - Modifies `Dockerfile`, `wait4.config.js` and CircleCI files to run containerized `auth-service` along with `MySQL` for handler tests

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,6 +18,6 @@ RUN adduser -D as-user
 USER as-user
 COPY --chown=as-user --from=builder /opt/auth-service .
 # cleanup
-RUN npm prune --production
+# RUN npm prune --production
```
Comment (eoln) [inline Dockerfile:21]:
```suggestion
# disable until problems with TS->JS transpilation solved
# RUN npm prune --production
```

**Comment by eoln** [comment]
@ramanmangla there is not necessary debugging logging in integration tests console, please clean it up

File: scripts/wait4.config.js [hunk 2]
```diff
@@ -22,6 +22,13 @@ module.exports = {
           uri: 'localhost:3306',
           method: 'mysql',
           retries: 30
+        },
+        {
+          description: 'Auth Service',
+          /* Change host:port accordingly based on default.json attributbes. */
```
Comment (eoln) [inline scripts/wait4.config.js:28]: wait4.config.js is Javascript file -> you can import default.json and pick up from it the host&port

File: test/integration/README.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# auth-service/test/integration #
+> all source code files are located in [/src](../src/README.md) folder
+
+## Running Integration Tests ##
+
+- Run `docker-compose build` to build the MySQL and auth-service images
+- Run `docker-compose up` to run the built docker images
+- Run `npm run test:integration` to run the integration tests
```
Comment (eoln) [inline test/integration/README.md:8]:
```suggestion
- Run `npm run test:integration` to run the integration tests (in a separate terminal session)
```

File: test/integration/server/handlers/consents.test.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+import axios from 'axios'
+// import { Consent } from '~/model/consent'
```
Comment (eoln) [inline test/integration/server/handlers/consents.test.ts:31]: drop commented code

File: test/integration/server/handlers/consents.test.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+describe('server/handlers/consents', (): void => {
+  it('Should return 202 (Accepted) status code',
+    async (): Promise<void> => {
+      // const consent: Consent = {
```
Comment (eoln) [inline test/integration/server/handlers/consents.test.ts:36]: drop commented code

File: test/integration/server/handlers/consents.test.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+          status: 'ACTIVE',
+          challenge: {
+            payload: 'base64(...)',
+            signature: 'base64(...)'
+          },
+          payload: 'base64(...)'
+        }
+      }
+
+      // fsioip encryption, signature, uri, http method
```
Comment (eoln) [inline test/integration/server/handlers/consents.test.ts:68]: what's for this comment?

File: test/integration/server/handlers/consents.test.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+      // x forwarded for
+
+      const headers = {
+        date: new Date().toJSON(),
+        'fspiop-source': 'third-party API',
+        'fspiop-destination': 'auth-service'
+      }
+
+      // let response: AxiosResponse
```
Comment (eoln) [inline test/integration/server/handlers/consents.test.ts:77]: drop commented code

File: test/integration/server/handlers/consents.test.ts [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+ --------------
+ ******/
+
+import axios from 'axios'
+
+describe('server/handlers/consents', (): void => {
+  it('Should return 202 (Accepted) status code',
+    async (): Promise<void> => {
+      // Endpoint
+      const scenariosURI = 'http://0.0.0.0:4004/consents'
```
Comment (eoln) [inline test/integration/server/handlers/consents.test.ts:36]:
```suggestion
      const scenariosURI = 'http://localhost:4004/consents'
```

File: test/integration/server/handlers/consents/{ID}/generateChallenge.test.ts [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+
+import axios from 'axios'
+
+describe('server/handlers/consents/{ID}/generateChallenge', (): void => {
+  it('Should return 202 (Accepted) status code',
+    async (): Promise<void> => {
+      const consentId = 'e3488c3a-a4f3-25a7-aa7a-fdc3994bb3ec'
+
+      // Endpoint
+      const scenariosURI = `http://0.0.0.0:4004/consents/${consentId}/generateChallenge`
```
Comment (eoln) [inline test/integration/server/handlers/consents/{ID}/generateChallenge.test.ts:38]:
```suggestion
      const scenariosURI = `http://localhost:4004/consents/${consentId}/generateChallenge`
```

File: test/integration/server/handlers/consents/{ID}/generateChallenge.test.ts [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+        'fspiop-destination': 'auth-service'
+      }
+
+      // try {
+      const response = await axios.post(scenariosURI, body, {
+        headers: headers
+      })
+
+      expect(response.status).toEqual(202)
+      // } catch (err) {
```
Comment (eoln) [inline test/integration/server/handlers/consents/{ID}/generateChallenge.test.ts:57]: drop commented out code

File: test/integration/server/handlers/thirdpartyRequests/transactions/{ID}/authorizations.test.ts [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+  (): void => {
+    it('Should return 202 (Accepted) status code',
+      async (): Promise<void> => {
+        const transactionId = '179395e8-8dd7-16a0-99f9-0da8f0c51c7f'
+
+        // TODO: URI should be
+        // `/thirdpartyRequests/transactions/{ID}/authorizations`
+        // Test needs to be changed once OpenAPI spec is updated
+        // in Ticket #412.
+        const scenariosURI = `http://0.0.0.0:4004/thirdPartyRequests/transactions/${transactionId}/authorizations`
```
Comment (eoln) [inline test/integration/server/handlers/thirdpartyRequests/transactions/{ID}/authorizations.test.ts:42]:
```suggestion
        const scenariosURI = `http://localhost:4004/thirdPartyRequests/transactions/${transactionId}/authorizations`
```

**Comment by lewisdaly** [comment]
outdated

**Pull request closed without merging**